core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -323,14 +348,29 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     }
   }
 
+  def updateCommitRecordMetadataForPendingTxnOffsetCommit(producerId: Long, topicPartition: TopicPartition,
```

Comment:
nit: this name is a bit unwieldy. Maybe it could be `onTxnOffsetCommitAppend` or something like that?